### PR TITLE
Create course version on unit group save

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Also add more protections to prevent changing the course version of unit group/scripts that have resources or vocab. 

Creating a course version on script save is already done [here](https://github.com/code-dot-org/code-dot-org/blob/5b459c38b4581bcdd6a8a73ce57c500e3fd9719e/dashboard/app/models/script.rb#L1077).

## Testing story

added a couple unit tests

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/lib/services/script_seed_test.rb [hunk 1]
```diff
@@ -1121,6 +1121,7 @@ def create_script_tree(
           create :lesson, lesson_group: lg, script: script, name: name, key: name, overview: "overview #{m + 1} #{n + 1}", has_lesson_plan: true
         end
       end
+      script.reload
 
       (1..num_resources_per_script).each do |r|
         resource = create :resource, key: "#{script.name}-resource-#{r}", course_version: course_version
```
Comment (davidsbailey) [inline dashboard/test/lib/services/script_seed_test.rb]: just noticing... should the blocks that create resources here be moved below `next unless course_version`?

**Review by davidsbailey — APPROVED** [review]
nice improvement!

File: dashboard/test/lib/services/script_seed_test.rb (hunk 1 above)
Comment (bethanyaconnor) [reply dashboard/test/lib/services/script_seed_test.rb]: oh totally! I'll add that in this PR as it shouldn't be a functional change

Comment (bethanyaconnor) [reply dashboard/test/lib/services/script_seed_test.rb]: hm actually, that `next unless course_version` is in the loop to create lessons, which isn't where these belong. I'll just add a specific check around these blocks

Comment (davidsbailey) [reply dashboard/test/lib/services/script_seed_test.rb]: ah, good catch. updates look good!

**Pull request merged**